src/backend/services/agent.py

```diff
@@ -42,3 +46,25 @@ def raise_db_error(e: Exception, type: str, name: str):
         )
 
     raise HTTPException(status_code=500, detail=str(e))
+
+
+def parse_task(t: SyncCeleryTaskMeta) -> AgentTaskResponse:
+    result = None
+    exception_snippet = None
+    if t.status == "SUCCESS":
+        result = pickle.loads(t.result)
+    if t.status == "FAILURE":
+        trace_lines = t.traceback.split("\n")
+        if len(trace_lines) >= 2:
+            # first 200 characters of the exception
+            exception_snippet = trace_lines[-2][:200] + "...check logs for details"
```

Comment:
Suggestion: move the `200` to some `TASK_EXCEPTION_MAX_LENGTH` (or any name that fits) constant at the top of this file